pkg/generate/code/converter/converter.go

```diff
@@ -0,0 +1,718 @@
+	out := ""
+	indent := strings.Repeat("\t", c.indentLevel)
+
+	switch shape.Type {
+	case "boolean", "string", "character", "byte", "short",
+		"integer", "long", "float", "double", "timestamp":
+		out += fmt.Sprintf(
+			// dst.Spec.Conditions = src.Spec.Conditions
+			"%s%s = %s\n",
+			indent, c.dstVar, c.srcVar,
+		)
+	default:
+		panic("Unsupported shape type: " + shape.Type)
+	}
+	return out
+}
+
+// copyList outputs Go code that copies one array to another.
+//
+// Output code will look something like this:
+//
+//  if src.Spec.APIStages != nil {
+//  	listOfAPIStageCopy := make([]*APIStage, 0, len(src.Spec.APIStages))
+//  	for i, element := range src.Spec.APIStages {
+//  		_ = i // non-used value guard.
+//  		elementCopy := &APIStage{}
+//  		if element != nil {
+//  			apiStageCopy := &APIStage{}
+//  			apiStageCopy.APIID = element.APIID
+//  			apiStageCopy.Stage = element.Stage
+//  			if element.Throttle != nil {
+//  				mapOfAPIStageThrottleSettingsCopy := make(map[string]*ThrottleSettings, len(element.Throttle))
+//  				for k, v := range element.Throttle {
+//  					elementCopy := &ThrottleSettings{}
+//  					if v != nil {
+//  						throttleSettingsCopy := &ThrottleSettings{}
+//  						throttleSettingsCopy.BurstLimit = v.BurstLimit
+//  						throttleSettingsCopy.RateLimit = v.RateLimit
+//  						elementCopy = throttleSettingsCopy
+//  					}
+//
+//  					mapOfAPIStageThrottleSettingsCopy[k] = elementCopy
+//  				}
+//  				apiStageCopy.Throttle = mapOfAPIStageThrottleSettingsCopy
+//  			}
+//
+//  			elementCopy = apiStageCopy
+//  		}
+//
+//  		listOfAPIStageCopy = append(listOfAPIStageCopy, elementCopy)
+//  	}
+//  	dst.Spec.APIStages = listOfAPIStageCopy
+//  }
+func (c converter) copyList(shape *awssdkmodel.Shape) string {
+	indent := strings.Repeat("\t", c.indentLevel)
+	// if a slice is only made of builtin types we just copy src to dst
+	if isMadeOfBuiltinTypes(shape) {
+		return fmt.Sprintf(
+			"%s%s = %s\n",
+			indent,
+			c.srcVar,
+			c.dstVar,
+		)
+	}
+
+	out := ""
+	// if err != nil {
+	out += fmt.Sprintf(
+		"%sif %s != nil {\n",
+		indent,
+		c.srcVar,
+	)
+
+	structShapeName := names.New(shape.ShapeName)
+	varStructCopy := structShapeName.CamelLower + "Copy"
+
+	// filterGroupsCopy := make([][]*v1alpha2.WebhookFilter, 0, len(src.Spec.FilterGroups))
+	out += c.withIndentLevel(c.indentLevel+1).newShapeTypeInstance(
+		shape,
+		varStructCopy,
+		c.srcVar,
+		false,
+	)
+
+	varIndex := "i"
+	varElement := "element"
+	// for i, element := range src.Spec.FilterGroups {
+	out += fmt.Sprintf(
+		"%s\tfor %s, %s := range %s {\n",
+		indent,
+		varIndex,
+		varElement,
+		c.srcVar,
+	)
+
+	// _ = i // non-used value guard.
+	out += fmt.Sprintf(
+		"%s\t\t_ = %s // non-used value guard.\n",
+		indent,
+		varIndex,
+	)
+
+	memberShapeRef := shape.MemberRef
+	memberShape := memberShapeRef.Shape
+	// elementCopy := make([]*v1alpha2.WebhookFilter, 0, len(element))
+	varElementCopy := "element" + "Copy"
```

Comment:
You are going to run into problems with nested list types here, because you will have variable name shadowing in effect. This is why the code generation that sets CR fields from an Output shape (the `pkg/generate/code.SetOutput` stuff) generates a variable name that represents the field's numeric index within the Shape. For instance, if I had a Shape like this:

```go
type Bar struct {
     Baz string
}

type Foo struct {
  Bars []*Bar
}

type MyShape struct {
  Foos []*Foo
}
```

The variable referring to `MyShape.Foos.Bars` would be called `f0f0` and the iterator over elements in that list of `Bar` structs would be called `f0f0iter`, with the element variable name of `f0f0elem`.

The way you are naming things here means that you would end up producing Go code like this:

```go
elementCopy := make([]*Foo, 0, len(src.Foos))
for _, element := range src.Foos {
    fooCopy := &Foo{}
    elementCopy := make([]*Bar, 0, len(element.Bars)
    for _, element := range element.Bars {
        barCopy := &Bar{}
        barCopy.Baz = element.Baz
        elementCopy = append(elementCopy, barCopy)
    }
    elementCopy = append(elementCopy, fooCopy)
}
```

which would be invalid since the variable name `elementCopy` is shadowed/hidden by the inner declaration of itself.